### PR TITLE
e2e: mysqlcluster-related api and test case.

### PR DESCRIPTION
**Pull request by runkecheng**
### What type of PR is this?

/enhancement

### Which issue(s) this PR fixes?

Ref: #302 

Fixes #

### What this PR does?

Summary:

- e2e framework(Adjust part of the code make the frame automatically export the log when the test fails)
- cluster scale in/out case

### CODE REVIEW

File: test/e2e/framework/portforward/portforward.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021 RadonDB.
```
Comment (andyli029) [inline test/e2e/framework/portforward/portforward.go:2]: Fix the line.

File: test/e2e/framework/test_context.go [hunk 2]
```diff
@@ -18,82 +18,56 @@ package framework
...
+	// Turn on verbose by default to get spec names
+	config.DefaultReporterConfig.Verbose = true
+
+	// Turn on EmitSpecProgress to get spec progress (especially on interrupt)
+	config.GinkgoConfig.EmitSpecProgress = true
+
+	// Randomize specs as well as suites
+	config.GinkgoConfig.RandomizeAllSpecs = true
+
+	flag.StringVar(&TestContext.KubeHost, "kubernetes-host", "", "The kubernetes host, or apiserver, to connect to")
```
Comment (acekingke) [inline test/e2e/framework/test_context.go:57]: use `fmt.Sprintf("%s:%s", os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")) ` may be better

File: test/e2e/framework/sysbench.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		do %s \
+		run; \
+		done`, cmd)
+	args := []string{cmd}
+	By("create sysbench pod for testing oltp")
+	pod, err := f.createSysbenchPod(mysqlcluster.Namespace, name, args)
+	Expect(err).Should(BeNil())
+
+	By("wait pod ready")
+	f.WaitPodReady(pod)
```
Comment (acekingke) [inline test/e2e/framework/sysbench.go:114]: Do not need wait ready，just wait finished

File: test/e2e/e2e.go [hunk 4]
```diff
@@ -17,9 +17,134 @@ limitations under the License.
...
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/radondb/radondb-mysql-kubernetes/test/e2e/framework"
+	"github.com/radondb/radondb-mysql-kubernetes/test/e2e/framework/ginkgowrapper"
 )
 
+const (
+	operatorNamespace = "mysql-operator"
+	releaseName       = "demo"
```
Comment (acekingke) [inline test/e2e/e2e.go:45]: use random name may be better

File: test/e2e/framework/helm.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	Expect(cmd.Run()).Should(Succeed())
+}
+
+func HelmPurgeRelease(release, ns string) {
+	args := []string{
+		"delete", release,
+		"--namespace", ns,
+		"--kube-context", TestContext.KubeContext,
+	}
+	cmd := exec.Command("helm", args...)
```
Comment (acekingke) [inline test/e2e/framework/helm.go:49]: In Makefile， should check `helm` exists,   and install it

**Review by acekingke — CHANGES_REQUESTED** [review]

File: test/e2e/framework/test_context.go (hunk 2 above)
Comment (runkecheng) [reply test/e2e/framework/test_context.go:57]: Kubeconfig contains Kubehost configuration, so when the remote connection k8s, the Host in kubeconfig can be used directly, no need to fill in the kubehost parameters.

File: test/e2e/framework/helm.go (hunk 5 above)
Comment (runkecheng) [reply test/e2e/framework/helm.go:49]: I will add a document to prompt this.

**Comment by acekingke** [comment]
LGTM

**Comment by andyli029** [comment]
Link: https://github.com/radondb/radondb-mysql-kubernetes/pull/302

**Pull request closed without merging**